ObjCVoiceCallKitQuickstart/ViewController.m

```diff
@@ -200,29 +200,32 @@ - (void)callDidConnect:(TVOCall *)call {
     [self routeAudioToSpeaker];
 }
 
-- (void)callDidDisconnect:(TVOCall *)call {
-    NSLog(@"callDidDisconnect:");
-
-    [self performEndCallActionWithUUID:call.uuid];
-
-    self.call = nil;
-    self.callKitCompletionCallback = nil;
+- (void)call:(TVOCall *)call didFailToConnectWithError:(NSError *)error {
+    NSLog(@"Call failed to connect: %@", error);
     
-    [self.placeCallButton setTitle:@"Place Outgoing Call" forState:UIControlStateNormal];
-    
-    [self toggleUIState:YES];
+    [self performEndCallActionWithUUID:call.uuid];
+    [self callDisconnected];
 }
 
-- (void)call:(TVOCall *)call didFailWithError:(NSError *)error {
-    NSLog(@"call:didFailWithError: %@", [error localizedDescription]);
-
+- (void)call:(TVOCall *)call didDisconnectWithError:(NSError *)error {
+    if (error) {
+        NSLog(@"Call failed: %@", error);
+        self.callKitCompletionCallback(NO);
```

Comment:
No need to fire the callback here, we are expecting a `didConnect` before a `didDisconnect`.